lasagne/utils.py

```diff
@@ -321,14 +322,35 @@ def create_param(spec, shape, name=None):
     This function is called by :meth:`Layer.add_param()` in the constructor
     of most :class:`Layer` subclasses. This enables those layers to
     support initialization with numpy arrays, existing Theano shared variables
-    or expressions, and callables for generating initial parameter values.
+    or expressions, and callables for generating initial parameter values,
+    Theano expressions, or shared variables.
     """
+    # to check if argument is a number
+    import numbers
     shape = tuple(shape)  # convert to tuple if needed
     if any(d <= 0 for d in shape):
         raise ValueError((
             "Cannot create param with a non-positive shape dimension. "
             "Tried to create param with shape=%r, name=%r") % (shape, name))
 
+    err_prefix = "cannot initialize parameter %s: " % name
+    if callable(spec):
+        spec = spec(shape)
+        err_prefix += "the %s returned by the provided callable"
+    else:
+        err_prefix += "the provided %s"
+
+    if isinstance(spec, numbers.Number):
+        spec = np.asarray(spec)
```

Comment:
> (I'm sorry that this PR turns out to be much more intricate than it seemed at first glance)

No, it's ok. Sorry that I have been slow in implementing the correct code. This is my first PR for Lasagne 😄 

> ```
> >>> x = np.float32(123)
> >>> isinstance(spec, numbers.Number)
> False
> ```

@f0k I think this code isn't completely correct. `numbers.Number` in fact correctly handles the case you have provided. 
```python
>>> x = np.float32(123)
>>> isinstance(x, numbers.Number)
True
```

As far as I tested multiple cases, `numbers.Number` correctly handles scalars of multiple types and kinds.